### PR TITLE
changed: Refactor + improve logging in CCurlFile

### PR DESCRIPTION
**Pull request by arnova**
This cleans up + improves the logging in CCurlFile to make it more consistent and easier to read CurlFile log entries.

### CODE REVIEW

File: xbmc/filesystem/CurlFile.cpp [hunk 1]
```diff
@@ -1100,8 +1108,8 @@ bool CCurlFile::Open(const CURL& url)
       ReadString(&error[0], 4095);
     }
 
-    CLog::Log(LOGERROR, "CCurlFile::Open failed with code {} for {}:\n{}", m_httpresponse,
-              redactPath, error);
+    CLog::Log(LOGERROR, "CCurlFile::{} - <{}> Failed with code {} for {}:\n{}", __FUNCTION__,
+              redactPath, m_httpresponse, error);
```
Comment (wsnipex) [inline xbmc/filesystem/CurlFile.cpp:1112]: I count 5 {}, but only 4 params

**Review by wsnipex — APPROVED** [review]
looks good, part for the minor

**Comment by Montellese** [comment]
Two hints:
* There is `CLog::LogFunction()` which expects the function name between the log level and the log message and automatically formats the function name into the log message. That should unify the logging with function name with other code parts using the same function. There's even the `CLog::LogF()` macro which uses `__FUNCTION__` to automatically fill in the function name.
* You could also retrieve a named logger using `CLog::GetLogger("CCurlFile")` and then use it directly to log stuff without having to specify `CCurlFile` as part of every log message.

**Comment by arnova** [comment]
> Two hints:
> 
> * There is `CLog::LogFunction()` which expects the function name between the log level and the log message and automatically formats the function name into the log message. That should unify the logging with function name with other code parts using the same function. There's even the `CLog::LogF()` macro which uses `__FUNCTION__` to automatically fill in the function name.
> * You could also retrieve a named logger using `CLog::GetLogger("CCurlFile")` and then use it directly to log stuff without having to specify `CCurlFile` as part of every log message.

Thanks for the hints! I wasn't aware of that. I'll incorporate it in this PR.

**Comment by arnova** [comment]
@Montellese : Decided to keep things as is for now. Using a named logger seemed nice but also makes things messy since we have 2 classes (CCurlFile & CReadState) in CurlFile.cpp.

**Comment by Montellese** [comment]
You can use the same named logger multiple times. I'll take a look later.

**Comment by arnova** [comment]
> You can use the same named logger multiple times. I'll take a look later.

Thanks @Montellese . If we can work out a proper solution, I'd like to use it for CFileCache as well.....

**Pull request merged**